src/lib/transaction_snark/transaction_snark.ml

```diff
@@ -3723,130 +3723,206 @@ let constraint_system_digests ~constraint_constants () =
     pair.
 *)
 let group_by_parties_rev partiess stmtss =
+  let use_full_commitment (p : Party.t) =
+    match p.authorization with
+    | Proof _ ->
+        `Proved_use_full_commitment p.data.body.use_full_commitment
+    | _ ->
+        `Others
+  in
   let rec group_by_parties_rev partiess stmtss acc =
     match (partiess, stmtss) with
     | ([] | [ [] ]), [ _ ] ->
         (* We've associated statements with all given parties. *)
         acc
-    | [ [ { Party.authorization = a1; _ } ] ], [ [ before; after ] ] ->
+    | [ [ ({ Party.authorization = a1; _ } as p) ] ], [ [ before; after ] ] ->
         (* There are no later parties to pair this one with. Prove it on its
            own.
         *)
-        (`Same, Parties_segment.Basic.of_controls [ a1 ], before, after) :: acc
-    | [ []; [ { Party.authorization = a1; _ } ] ], [ [ _ ]; [ before; after ] ]
-      ->
+        ( `Same
```

Comment:
Consider defining a record type in a local module to give names to all these fields-- I think it would clear up what's happening here